addons/mail/models/mail_tracking_value.py

```diff
@@ -31,6 +31,9 @@ class MailTracking(models.Model):
     new_value_text = fields.Text('New Value Text', readonly=1)
     new_value_datetime = fields.Datetime('New Value Datetime', readonly=1)
 
+    currency_id = fields.Many2one('res.currency', 'Currency', readonly=1, ondelete='set null',
```

Comment:
Do we actually want to support deleting a currency that was previously used in the system? Seems like a dangerous idea.

Reply:
Tracking value should not prevent generic behavior. If we want to avoid removing currencies this should be managed at currency level, not here. I don't know what is the behavior of currencies and I think tracking should not mess with it. I think setting null if removed is safe enough as its purpose is simply to continue working whatever happens to the line in currency table.

Reply:
Maybe whatever happens at currency code it relying on checking if the currency is used anywhere by trying to delete it and see if there is any restrict that is raised?

I don't really care what we decide as long as we at least thought about it.

Reply:
I wouldn't raise here. Tracking still works without currency as this is a nice to have.